SourceryTests/Parsing/FileParserSpec.swift

```diff
@@ -153,6 +163,35 @@ class FileParserSpec: QuickSpec {
                         expect(parse("// sourcery: thirdLine = 4543\n/// comment\n// sourcery: firstLine\nclass Foo: TestProtocol { }"))
                                 .to(equal([expectedType]))
                     }
+
+                    it("extracts generics correctly") {
+                        let result = parse("""
+class Foo<A : Equatable & Codable, B : Baz<A>,C> : Bar<B> \nwhere C : CustomStringConvertible, \nB : Comparable & Decodable, \n\nC : CustomDebugStringConvertible {
+    func f<T : Codable>(_ t : T) -> T where T : Equatable {
+        return t
+    }
+}
+""")
+                        let expected = Class(name: "Foo", isGeneric: true, genericTypeParameters: [GenericTypeParameter(typeName: TypeName("A"), constraints: [GenericTypeParameterConstraint(name: TypeName("Equatable")), GenericTypeParameterConstraint(name: TypeName("Codable"))]), GenericTypeParameter(typeName: TypeName("B"), constraints: [GenericTypeParameterConstraint(name: TypeName("Baz<A>")), GenericTypeParameterConstraint(name: TypeName("Comparable")), GenericTypeParameterConstraint(name: TypeName("Decodable"))]), GenericTypeParameter(typeName: TypeName("C"), constraints: [GenericTypeParameterConstraint(name: TypeName("CustomStringConvertible")), GenericTypeParameterConstraint(name: TypeName("CustomDebugStringConvertible"))])])
+                        let method = Method(name: "f<T : Codable>(_ t : T)", selectorName: "f(_:)", parameters: [MethodParameter(argumentLabel: nil, name: "t", typeName: TypeName("T"), type: nil, defaultValue: nil, annotations: [:], isInout: false)], returnTypeName: TypeName("T where T : Equatable"), throws: false, rethrows: false, accessLevel: .internal, isStatic: false, isClass: false, isFailableInitializer: false, definedInTypeName: TypeName("Foo"), genericTypeParameters: [GenericTypeParameter(typeName: TypeName("T"), constraints: [GenericTypeParameterConstraint(name: TypeName("Codable")), GenericTypeParameterConstraint(name: TypeName("Equatable"))])])
+                        expected.inheritedTypes = ["Bar<B>"]
+                        expected.methods.append(method)
+
+                        expect(result).to(equal([expected]))
+                    }
+
+                    it("doesn't extract generics if they are concrete parameters to the supertype") {
+                        let result = parse("class Foo : Bar<Int> {}")
+                        let expected = Class(name: "Foo", isGeneric: false)
+                        expected.inheritedTypes = ["Bar<Int>"]
```

Comment:
ideally the generic type information should be still available in this case. In the end if `Boo` is defined as `Boo<Value>`, then `Foo.Value` can be used in the `Foo` implementation. And it should have a concrete type information too.